### PR TITLE
[Example] with-typescript-graphql-hooks-styled-components example

### PR DESCRIPTION
**Pull request by ManAnRuck**
add an example which combines

typescript
graphql-hooks
styled-components

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -28,3 +28,4 @@ test/**/next-env.d.ts
 # example output
 examples/**/out
 
+/.vscode
```
Comment (lfades) [inline .gitignore]: This is a change for another PR, let's remove it

**Comment by lfades** [comment]
@ManAnRuck Hi thank you for the PR, here's some feedback from my quick view over the files:

- Exclude automatically generated files
- You're adding `styled-components` but using `styled-jsx`, only use one. If you decide for styled-jsx then rename the example to be `with-graphql-hooks-typescript`
- if you can remove Apollo related packages and only have `graphql-hooks`, please do.
- Remove packages that are not necessary for the example to work, like all the eslint packages

**Review by lfades — CHANGES_REQUESTED** [review]
Check out my comments 🕵️‍♂️

**Comment by ManAnRuck** [comment]
sure, I will make the requested changes.

removing styled-components
remove generated files (and add a script to generate them)
remove eslint

I don't want to remove typescript in reason of there is an js Apollo-hooks example (but missing with typescript)

**Comment by ManAnRuck** [comment]
this PR is now deprecated in reason of the official [apollo hooks](https://www.npmjs.com/package/@apollo/react-hooks)

**Pull request closed without merging**